sigal/gallery.py

```diff
@@ -290,17 +293,25 @@ def __init__(self, path, settings, dirnames, filenames, gallery):
         self.medias = medias = []
         self.medias_count = defaultdict(int)
 
-        for f in filenames:
-            ext = splitext(f)[1]
-            if ext.lower() in settings['img_extensions']:
-                media = Image(f, self.path, settings)
-            elif ext.lower() in settings['video_extensions']:
-                media = Video(f, self.path, settings)
+        if filenames:
+            if gallery.pool:
+                # Chunksize computation taken from Pool.map code
+                chunksize, extra = divmod(len(filenames), len(gallery.pool._pool) * 4)
+                if extra or chunksize < 1:
+                    chunksize += 1
+                results = gallery.pool.imap_unordered(
+                        album_worker,
+                        ((self.path, f, settings) for f in filenames),
+                        chunksize)
```

Comment:
Reusing the gallery pool seems a bit fragile, not sure if this is a good idea.

Reply:
It's not as fragile as the code is _right now_. All albums are collected before the actual processing, so the pool should be free when we need it again.